File: DevNews/DevNews.Server/Article.Web.Server/routes/client/article.ts
import * as express from 'express'
import { Request, Response } from 'express-serve-static-core'
import { UpsertArticle } from '../../model/article'
import IArticle from '../../service/rule/IArticle'
import Article from '../../service/service/Article'
const router = express.Router()
const _article: IArticle = new Article()

router.post("/Create", async (req: Request, res: Response) => {
    let body = req.body
    let article: UpsertArticle = {
        title: body.title,
        text: body.text,
        shortDescription: body.shortDescription,
        file: body.file,
        attachments: body.attachments,
        channelToken: body.channelToken,
        isActive: body.isActive,
        ownerType: body.ownerType,
        tags: body.tags
    }
    let create = await _article.createArticle(article, req.headers)
    res.json(create)
    res.end()
})

export default router

File: DevNews/Entity/Article/Article.cs
namespace Entity.Article;

public class Article
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public string Title { get; set; }

    [Required]
    public string ShortDescription { get; set; }

    [Required]
    public string Text { get; set; }

    public string Tags { get; set; }

    [Required]
    public Guid OwnerId { get; set; }

    [Required]
    public short OwnerType { get; set; }


    public virtual ICollection<ArticleCategories> ArticleCategories { get; set; }

}
